### PR TITLE
Appoint a Rep component unit tests

### PR DESCRIPTION
**Pull request by cosu419**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No

## Summary

Unit tests for components:
- SelectAccreditedRepresentative
- SelectOrganization
- ContactAccreditedRepresentative
- ClaimantType
- SubmissionError

## Related issue(s)

- https://github.com/department-of-veterans-affairs/vets-website/pull/33325

## Testing done

- Unit tests covering our component folder

### CODE REVIEW

File: src/applications/representative-appoint/components/SelectAccreditedRepresentative.jsx [hunk 1]
```diff
@@ -111,7 +111,7 @@ const SelectAccreditedRepresentative = props => {
   };
 
   const handleSelectRepresentative = async selectedRepResult => {
-    if (selectedRepResult === currentSelectedRep) {
+    if (selectedRepResult === currentSelectedRep && isReviewPage) {
```
Comment (jvcAdHoc) [inline src/applications/representative-appoint/components/SelectAccreditedRepresentative.jsx:114]:
```js
  const currentSelectedRep = useRef(
    formData?.['view:representativeSearchResults'],
  );
```

Would `currentSelectedRep` always be an array of search results? If so, this conditional could never be true.

File: src/applications/representative-appoint/tests/components/SubmissionError.unit.spec.jsx
Comment (jvcAdHoc) [inline src/applications/representative-appoint/tests/components/SubmissionError.unit.spec.jsx:1]: Might be good to separate this into multiple tests based on behavior: one for what happens on first error and one for what happens on rerender

File: src/applications/representative-appoint/tests/pages/representative/contactAccreditedRepresentative.unit.spec.jsx
Comment (jvcAdHoc) [inline src/applications/representative-appoint/tests/pages/representative/contactAccreditedRepresentative.unit.spec.jsx:1]: test organization nit: I think these tests would be easier to read if grouped into contexts.

Along these lines:
```
context('when in review mode')
  context('when navigating backwards')
    it('does the thing')
  context('when navigating forwards')
context('when not in review mode')
...
```

File: src/applications/representative-appoint/tests/pages/representative/selectAccreditedRepresentative.unit.spec.jsx
Comment (jvcAdHoc) [inline src/applications/representative-appoint/tests/pages/representative/selectAccreditedRepresentative.unit.spec.jsx:1]: similar thoughts on organization. Also, a test for the `if (selectedRepResult === currentSelectedRep && isReviewPage)` routing would be good to have.

File: src/applications/representative-appoint/components/SelectAccreditedRepresentative.jsx (hunk 1 above)
Comment (cosu419) [reply src/applications/representative-appoint/components/SelectAccreditedRepresentative.jsx]: Should be fixed.

**Review by jvcAdHoc — APPROVED** [review]

**Pull request merged**